scripts/fast-reboot

```diff
@@ -15,9 +15,10 @@ then
 fi
 
 # Kernel and initrd image
-KERNEL_IMAGE="/vmlinuz"
-INITRD="/initrd.img"
-BOOT_OPTIONS=$(cat /proc/cmdline)
+NEXT_SONIC_IMAGE=$(sonic_installer list | grep "Next: " | cut -d ' ' -f 2)
+BOOT_OPTIONS=$(cat /host/grub/grub.cfg | sed "/$NEXT_SONIC_IMAGE'/,/}/"'!'"g" | grep linux)
+KERNEL_IMAGE="/host$(echo $BOOT_OPTIONS | cut -d ' ' -f 2)"
+INITRD=$(echo $KERNEL_IMAGE | sed 's/vmlinuz/initrd.img/g')
 
 case "$BOOT_OPTIONS" in
```

Comment:
You can remove this block and append fast-reboot option unconditionally. You read the command line from the file

Reply:
agree